bears/general/InvalidLinkBear.py

```diff
@@ -43,8 +43,19 @@ def get_status_code(url, timeout):
     def find_links_in_file(file, timeout, link_ignore_regex):
         link_ignore_regex = re.compile(link_ignore_regex)
         regex = re.compile(
-            r'(https?://[^.:%\s_/?#[\]@\\]+\.(?:[^\s()%\'"`<>|\\]+|'
-            r'\([^\s()%\'"`<>|\\]*\))*)(?<!\.)(?<!,)')
+            r'(https?://'  # http:// or https://
+            r'[^.:%\s_/?#[\]@\\]+'  # Initial Part of Domain
+            r'\.'  # A required dot (.)
+            r'(?:[^\s()%\'\"`<>|\\]+'  # Domain / Subdomain Name
+            r'|'  # OR
+            r'\([^\s()%\'\"`<>|\\]*\))'
+            # Subdomain Name that is contained within ()
+            r'%?[0-9]*'  # A % sign (0 or 1 times) followed by any number
```

Comment:
Because according to what you guys want, I shall try to modify the regex although its getting complicated already.